### ISSUE TITLE
Understanding Multi-Line Text input for training and testing model

### ISSUE THREAD

**Issue by aashish2000** [drboog/Lafite#19]
Hi, from my understanding, `dataset_tool.py` supports datasets having more than one line of text caption per image but the testing code only allows for a single line to be passed as input per query. I would like to know how/where in the codebase these multiple captions lines are passed for training and if the inference code can be updated for testing with multi-line caption queries.

I am using the Ground-Truth pair based training approach for evaluation. Thanks in advance.

**Comment by drboog** [comment]
For evaluation, 30k images are randomly generated with respect to captions of validation set (MS-COCO validation set contains around 40k images and 200k captions, each image has 5 captions).
In the dataset file, 5 corresponding features will be stored for each image, and they will be randomly selected during testing. Please see _get_txt_features_ at _training/dataset.py_.
If you want to generate 5 images for all the 5 captions, which resulting in 200k images, you may want to write your own script and iterate over all the captions.
